sources/procfs.go

```diff
@@ -675,6 +675,31 @@ func parseFileElements(path string, directoryDepth int) (name string, nodeName s
 	return name, nodeName, nil
 }
 
+func convertToBytes(s string) string {
+	new_s := ""
+	multiplier := 1
+	switch finalChar := s[len(s)-1:]; finalChar {
```

Comment:
Another thought, are we sure the length of `s` will always be at least 1? I'm pretty sure we can be confident it is at least 1 as we shouldn't make it to this function if it isn't, but might not hurt to double check.